src/test/java/jenkins/plugins/gerrit/workflow/GerritReviewStepTest.java

```diff
@@ -142,6 +148,69 @@ public void gerritReviewStepInvokeFailSSLValidationTest() throws Exception {
     j.assertLogContains("javax.net.ssl.SSLHandshakeException", run);
   }
 
+  @Test
+  public void gerritReviewStepInvokeAnonymousTest() throws Exception {
```

Comment:
this test shows that the restapi client is able to work anonymously, there is no reason why we should not allow that.

Reply:
Actually, it wasn't working and that's why I added the test. When I find a bug I first write a test to demonstrate that the bug is there and then I fix it.

Without this test, you could not realize that the code (and the feature) is broken. Makes sense?